packages/grid/x-data-grid-pro/src/components/DataGridProVirtualScroller.tsx

```diff
@@ -119,18 +136,48 @@ const VirtualScrollerPinnedColumns = styled('div', {
   position: 'sticky',
   overflow: 'hidden',
   zIndex: 1,
-  boxShadow: theme.shadows[2],
   backgroundColor: theme.palette.background.default,
-  ...(theme.palette.mode === 'dark' && {
-    backgroundImage: `linear-gradient(${alpha('#fff', getOverlayAlpha(2))}, ${alpha(
-      '#fff',
-      getOverlayAlpha(2),
-    )})`,
+  ...(theme.palette.mode === 'dark' && { backgroundImage: darkModeBackgroundImage }),
+  ...(ownerState.side === GridPinnedPosition.left && {
+    left: 0,
+    float: 'left',
+    boxShadow: '2px 0px 4px -2px rgb(0 0 0 / 20%)',
```

Comment:
I had to use custom shadow instead of `theme.shadows` to only show shadow on one side of the pinned columns container, otherwise when using pinned rows + pinned columns shadows overlap and look ugly

Reply:
Maybe we can take the color from the theme?